.github/workflows/minimum-version-ts-check.yaml

```diff
@@ -0,0 +1,46 @@
+name: Library checks
+
+on:
+    pull_request:
+    push:
+        branches:
+            - main
+
+jobs:
+    unit:
+        name: Check minimum TS version
+        runs-on: ubuntu-latest
+        strategy:
+          matrix:
+            node-version: [ 16, 18, 20, 22, 23 ]
+            typescript-version: ['4.7.2', '5.5.4' ]
```

Comment:
Maybe you could do an explicit `latest` here instead? This way we don't need to keep updating this every time Microsoft decides they want to improve speed by 0.0001%